networkx/tests/test_convert_geopandas.py

```diff
@@ -0,0 +1,85 @@
+import os
+import pytest
+import networkx as nx
+
+gpd = pytest.importorskip("geopandas")
+
+dirname = os.path.dirname(__file__)
+
+
+def test_from_geopandas_edgelist_undirected():
+
+    gdf = gpd.GeoDataFrame.from_file(os.path.join(dirname, "test_shp.shp"))
+    graph_from_shp = nx.from_geopandas_edgelist(gdf, create_using=nx.Graph)
+    assert isinstance(graph_from_shp, nx.Graph)
+
+
+def test_from_geopandas_edgelist_directed():
+
+    gdf = gpd.GeoDataFrame.from_file(os.path.join(dirname, "test_shp.shp"))
+    graph_from_shp = nx.from_geopandas_edgelist(gdf, create_using=nx.DiGraph)
+    assert isinstance(graph_from_shp, nx.DiGraph)
+
+
+def test_from_geopandas_edgelist_undirected_with_attrs():
+
+    gdf = gpd.GeoDataFrame.from_file(os.path.join(dirname, "test_shp.shp"))
+    graph_from_shp = nx.from_geopandas_edgelist(
+        gdf, create_using=nx.Graph, edge_attr=gdf.columns.to_list()
+    )
+    failed = False
+    for u, v, d in graph_from_shp.edges(data=True):
+        if set(d.keys()) != {"id", "attr_text", "attr_float", "geometry", "attr_int"}:
+            failed = True
+            break
+    assert not failed
+
+
+def test_from_geopandas_edgelist_directed_with_attrs():
+
+    gdf = gpd.GeoDataFrame.from_file(os.path.join(dirname, "test_shp.shp"))
+    graph_from_shp = nx.from_geopandas_edgelist(
+        gdf, create_using=nx.DiGraph, edge_attr=gdf.columns.to_list()
+    )
+    failed = False
+    for u, v, d in graph_from_shp.edges(data=True):
+        if set(d.keys()) != {"id", "attr_text", "attr_float", "geometry", "attr_int"}:
+            failed = True
+            break
+    assert not failed
+
+
+def test_to_geopandas_edgelist():
```

Comment:
This test has the potential to check that ```to_geopandas_edgelist``` and ```from_geopandas_edgelist``` are inverses -- that is, the ```gdf``` we start with equals the ```new_gdf``` we end with after a round-trip into nx.
You check that the data is the same, but not that the edges are the same. 
Is there an easy: ```gdf == new_gdf``` capability in geopandas?  Something similar?